src/helpers/clipboard/clipboard_figure_state.ts

```diff
@@ -84,12 +84,20 @@ export class ClipboardFigureState implements ClipboardState {
    */
   paste(target: Zone[]) {
     const sheetId = this.getters.getActiveSheetId();
+    const size = { height: this.copiedFigure.height, width: this.copiedFigure.width };
+    const viewportRect = this.getters.getMainViewportRect();
```

Comment:
I don't think you are looking for this value but rather the size of the grid that can be seen by the client? because the "mainViewport" will have another size that the aforementioned when we have frozen panes. Specifically,  the 'Main viewport' corresponds to the bottom right viewport.  It just so happens that it is equal to the grid size when there are not frozen panes. But you should take those into account.

Reply:
what about this ?